From 3d7c588196566b0b981bef192800e91d3ddecfde Mon Sep 17 00:00:00 2001
From: Olivier Moysan <olivier.moysan@st.com>
Date: Mon, 11 Mar 2019 10:31:37 +0100
Subject: [PATCH] ASoC: wm8994: add LDO cycle-time protection

The minimum cycle time of wm8994 audio codec
LDOx regulator is specified at 36ms.
Add a 40ms delay on regulator enable at resume
to match this requirement.

Change-Id: I15803071a0e641e284d15a16a6056ceeeafd1254
Signed-off-by: Olivier Moysan <olivier.moysan@st.com>
---
 drivers/mfd/wm8994-core.c | 6 ++++++
 1 file changed, 6 insertions(+)

--- a/drivers/mfd/wm8994-core.c
+++ b/drivers/mfd/wm8994-core.c
@@ -184,6 +184,12 @@ static int wm8994_resume(struct device *
 	if (!wm8994->suspended)
 		return 0;
 
+	/*
+	 * LDO1/2 minimum cycle time is 36ms according to codec specification
+	 * Wait before enabling regulator to make sure we fit this requirement
+	 */
+	msleep(40);
+
 	ret = regulator_bulk_enable(wm8994->num_supplies,
 				    wm8994->supplies);
 	if (ret != 0) {
